### PR TITLE
Remove quarkus-properties language and collect document selectors from extensions

### LINKED ISSUE #3: Remove quarkus-properties language
The goal of this issue os to remove quarkus-properties language declaration in https://github.com/redhat-developer/vscode-microprofile/pull/1/files/ab9336fea779e140cf9dd4afa0521c016778d859#r455636547 which is used to

 *  bind quarkus application.properties to the MicroProfile LS
 * bind textmate grammar for application.properties file

We need to remove quarkus-properties and have just `microprofile-properties` language, but in vscode-quarkus we need `quarkus-properties`, so it means that:

 * vscode-microprofile must provide a documentSelector extensions to declare in `package.json` of some extensions (ex: vscode-quarkus) the following document selector:

```json
"contributes": 
  "microprofile":
     "documentSelector":  
[
  { scheme: 'file', language: 'quarkus-properties' }
]
```

After that we could remove https://github.com/redhat-developer/vscode-microprofile/pull/1#pullrequestreview-449666195 and search compute documentSelector my merging the microprofile documentSelector with documentSelector coming from extension (like vscode-quarkus)

 * how to bind the textmate grammar from vscode-microprofile to the quarkus-properties language coming from vscode-quarkus? @xorye have you some idea? When we will find a solution, we will able to to remove https://github.com/redhat-developer/vscode-microprofile/pull/1/files/ab9336fea779e140cf9dd4afa0521c016778d859#r455641270

### PR DESCRIPTION
**Pull request by xorye**
Fixes #3 

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

File: schemas/package.schema.json [hunk 1]
```diff
@@ -16,6 +16,19 @@
                 "type": "string",
                 "description": "Relative path to a MicroProfile language server extension JAR file"
               }
+            },
+            "documentSelector": {
+              "type": "object",
+              "properties": {
+                "scheme": {
+                  "type": "string",
+                  "description": "A language id, like `microprofile-properties`"
```
Comment (rzgry) [inline schemas/package.schema.json:25]: I think the descriptions are flipped for scheme / language

Comment (xorye) [reply schemas/package.schema.json]: thank you, fixed

**Comment by xorye** [comment]
> Looking at the schema It looks like this provides a vscode extension the ability to provide a single document selector. Would their ever be a use-case where an extension would need to provide multiple document selectors?

@rzgry thanks for the review, I will make changes.

Also, to be consistent with https://github.com/microsoft/vscode-languageserver-node/blob/3eddc56d9491b6f97c5e82242a3a14421d42bea1/protocol/src/common/protocol.ts#L74, I will add `pattern` as a possible field.

**Comment by xorye** [comment]
@rzgry @datho7561 if there aren't any more changes to be made, I'd like to merge this soon. Please let me know, thanks!

**Comment by xorye** [comment]
For some reason, the `client/registerCapability` request has `quarkus-properties` appearing twice:

MicroProfile Tools server trace:
```
[Trace - 11:16:26 a.m.] Received request 'client/registerCapability - (8)'.
Params: {
    "registrations": [
        {
            "id": "61e7b5ba-a567-4415-8817-70e18fed36a9",
            "method": "textDocument/rangeFormatting",
            "registerOptions": {
                "documentSelector": [
                    {
                        "language": "microprofile-properties"
                    },
                    {
                        "language": "quarkus-properties"
                    },
                    {
                        "language": "quarkus-properties"
                    }
                ]
            }
        }
    ]
}
```

I don't think its caused by this PR though. I'm investigating.

**Comment by xorye** [comment]
It's not caused by this PR, this still happens in the 1.6.0 vscode-quarkus release. I made an issue about this: https://github.com/redhat-developer/vscode-quarkus/issues/296

**Pull request merged**